sormas-backend/src/main/java/de/symeda/sormas/backend/deletionconfiguration/DeletionConfigurationService.java

```diff
@@ -0,0 +1,53 @@
+package de.symeda.sormas.backend.deletionconfiguration;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.ejb.LocalBean;
+import javax.ejb.Stateless;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import de.symeda.sormas.backend.common.BaseAdoService;
+
+@Stateless
+@LocalBean
+public class DeletionConfigurationService extends BaseAdoService<DeletionConfiguration> {
+
+
+	public DeletionConfigurationService() {
+		super(DeletionConfiguration.class);
+	}
+
+	public DeletionConfiguration getCoreEntityTypeConfig(CoreEntityType coreEntityType) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<DeletionConfiguration> cq = cb.createQuery(getElementClass());
+		Root<DeletionConfiguration> from = cq.from(getElementClass());
+		cq.where(cb.equal(from.get(DeletionConfiguration.ENTITY_TYPE), coreEntityType));
+
+		return em.createQuery(cq).getSingleResult();
+	}
+
+	public void createMissingDeletionConfiguration() {
```

Comment:
Not quite sure
a) how this is supposed to work since the `build` method doesn't set a reference and period, but both are supposed to be not nullable
b) whether we want something like this in the first place because in that case, we'd also need to define default values ourselves and I'm not sure whether that's a good idea (since countries might not want to use this feature at all, or might have other regulations than those we define)

Reply:
Initially, the implementation was to create a default DeleteConfiguration for each Core Entity. In the review notes above I received the recommendation to remove default deletion reference and deletion period. 
Now it works like this, on startup if a DeletetionConfiguration is missing of a CoreEntity it will be added have DeletionReference and DeletionPeriod null. When these 2 values are not defined the deletion code is not executed. This check is in CoreEntityDeletionService.